### ISSUE TITLE
QtWebEngine fails to build with NSS 3.23

### ISSUE THREAD

**Issue by tkemmer** [BALL-Project/ball_contrib#11]
QtWebEngine fails to build on systems with NSS >=3.23 as some struct members have been renamed.

```
FAILED: /usr/bin/gcc -MMD -MF obj/src/3rdparty/chromium/net/third_party/nss/ssl/libssl.ssl3con.o.d -DV8_DEPRECATION_WARNINGS -DTOOLKIT_QT -D_FILE_OFFSET_BITS=64 -DNO_TCMALLOC -DCHROMIUM_BUILD -DTOOLKIT_VIEWS=1 -DUI_COMPOSITOR_IMAGE_TRANSPORT -DUSE_AURA=1 -DUSE_DEFAULT_RENDER_THEME=1 -DUSE_LIBJPEG_TURBO=1 -DUSE_X11=1 -DENABLE_ONE_CLICK_SIGNIN -DENABLE_PRE_SYNC_BACKUP -DUSE_XI2_MT=2 -DENABLE_REMOTING=1 -DENABLE_WEBRTC=1 -DENABLE_PEPPER_CDMS -DENABLE_CONFIGURATION_POLICY -DENABLE_NOTIFICATIONS -DUSE_UDEV -DDONT_EMBED_BUILD_METADATA -DENABLE_TASK_MANAGER=1 -DENABLE_EXTENSIONS=1 -DENABLE_PLUGINS=1 -DENABLE_SESSION_SERVICE=1 -DENABLE_THEMES=1 -DENABLE_AUTOFILL_DIALOG=1 -DENABLE_BACKGROUND=1 -DENABLE_GOOGLE_NOW=1 -DCLD_VERSION=2 -DCLD2_DATA_SOURCE=static -DENABLE_PRINTING=1 -DENABLE_BASIC_PRINTING=1 -DENABLE_PRINT_PREVIEW=1 -DENABLE_SPELLCHECK=1 -DENABLE_CAPTIVE_PORTAL_DETECTION=1 -DENABLE_APP_LIST=1 -DENABLE_SETTINGS_APP=1 -DENABLE_MANAGED_USERS=1 -DENABLE_MDNS=1 -DENABLE_SERVICE_DISCOVERY=1 -DENABLE_LOAD_COMPLETION_HACKS=1 -DNO_PKCS11_BYPASS -DNSS_ENABLE_ECC -DUSE_UTIL_DIRECTLY -DUSE_LIBPCI=1 -DUSE_GLIB=1 -DUSE_NSS=1 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -Igen -I/home/kemmer/git/ballcontrib/build/src/qt/qtwebengine/src/3rdparty/chromium/net/third_party/nss/ssl/bodge -I/usr/include/libdrm -fstack-protector --param=ssp-buffer-size=4  -pthread -fno-strict-aliasing -Wno-unused-parameter -Wno-missing-field-initializers -fvisibility=hidden -pipe -fPIC -Wno-unused-local-typedefs -I/usr/include/nss -I/usr/include/nspr -Wno-format -Wno-unused-result -m64 -march=x86-64 -O2 -fno-ident -fdata-sections -ffunction-sections -funwind-tables   -c /home/kemmer/git/ballcontrib/build/src/qt/qtwebengine/src/3rdparty/chromium/net/third_party/nss/ssl/ssl3con.c -o obj/src/3rdparty/chromium/net/third_party/nss/ssl/libssl.ssl3con.o
/home/kemmer/git/ballcontrib/build/src/qt/qtwebengine/src/3rdparty/chromium/net/third_party/nss/ssl/ssl3con.c: In function ‘ssl3_ChaCha20Poly1305’:
/home/kemmer/git/ballcontrib/build/src/qt/qtwebengine/src/3rdparty/chromium/net/third_party/nss/ssl/ssl3con.c:2091:15: error: ‘CK_NSS_AEAD_PARAMS {aka struct CK_NSS_AEAD_PARAMS}’ has no member named ‘pIv’
     aeadParams.pIv = (unsigned char *) additionalData;
               ^
/home/kemmer/git/ballcontrib/build/src/qt/qtwebengine/src/3rdparty/chromium/net/third_party/nss/ssl/ssl3con.c:2092:15: error: ‘CK_NSS_AEAD_PARAMS {aka struct CK_NSS_AEAD_PARAMS}’ has no member named ‘ulIvLen’
     aeadParams.ulIvLen = 8;
               ^
ninja: build stopped: subcommand failed.
```

Right now, the source code can be patched manually on affected systems:
[patch-nss.txt](https://github.com/BALL-Project/ball_contrib/files/240591/patch-nss.txt)

**Comment by tkemmer** [comment]
Fixed with Qt 5.6.

**Comment by philthiel** [comment]
One reason for adding Qt 5.6.1 to ball_contrib : )
